src/main/kotlin/no/nav/arbeidsgiver/min_side/services/digisyfo/DigisyfoRepository.kt

```diff
@@ -0,0 +1,210 @@
+import io.micrometer.core.instrument.MeterRegistry
+import org.springframework.context.annotation.Profile
+import org.springframework.jdbc.core.JdbcTemplate
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Repository
+import java.sql.Date
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.time.LocalDate
+import java.time.ZoneId
+import java.util.concurrent.TimeUnit
+import java.util.function.Function
+import java.util.stream.Collector
+import java.util.stream.Collectors
+
+interface DigisyfoRepository {
+    data class Virksomhetsinfo(
+        val virksomhetsnummer: String = "",
+        val antallSykmeldte: Int = 0
+    )
+
+    fun virksomheterOgSykmeldte(nærmestelederFnr: String): List<Virksomhetsinfo>
+    fun processNærmesteLederEvent(hendelse: NarmesteLederHendelse)
+    fun processSykmeldingEvent(records: List<Pair<String?, SykmeldingHendelse?>>)
+}
+
+@Profile("dev-gcp", "prod-gcp")
+@Repository
+class DigisyfoRepositoryImpl(
+    private val jdbcTemplate: JdbcTemplate,
+    meterRegistry: MeterRegistry,
+) : DigisyfoRepository {
+    private val tombstoneCounter: Counter
+    private val expiredCounter: Counter
+    private val updateCounter: Counter
+
+    init {
+        tombstoneCounter = Counter.builder("fager.msa.sykmelding.tombstones")
+            .description("antall sykmelding-tombstones prosessert")
+            .register(meterRegistry)
+        expiredCounter = Counter.builder("fager.msa.sykmelding.expired")
+            .description("antall sykmeldinger som er slettet fordi de er utløpt")
+            .register(meterRegistry)
+        updateCounter = Counter.builder("fager.msa.sykmelding.updated")
+            .description("antall sykmeldinger som er opprettet/oppdatert")
+            .register(meterRegistry)
+    }
+
+    override fun processNærmesteLederEvent(hendelse: NarmesteLederHendelse) {
+        if (hendelse.aktivTom != null) {
+            jdbcTemplate.update("delete from naermeste_leder where id = ?") { ps: PreparedStatement ->
+                ps.setObject(1, hendelse.narmesteLederId)
+            }
+        } else {
+            jdbcTemplate.update(
+                """
+                insert into naermeste_leder(id, naermeste_leder_fnr, virksomhetsnummer, ansatt_fnr)  
+                values(?, ?, ?, ?)  
+                on conflict (id) 
+                do nothing;
+                """.trimIndent()
+            ) { ps: PreparedStatement ->
+                ps.setObject(1, hendelse.narmesteLederId)
+                ps.setString(2, hendelse.narmesteLederFnr)
+                ps.setString(3, hendelse.virksomhetsnummer)
+                ps.setString(4, hendelse.ansattFnr)
+            }
+        }
+    }
+
+    override fun processSykmeldingEvent(records: List<Pair<String?, SykmeldingHendelse?>>) {
+        val deduplicated = records.stream().collect(latestBy(Pair<String?, SykmeldingHendelse?>::first))
```

Comment:
@peterbb kan du ta en dobbelsjekk av logikk for deduplicated 
vi skulle kanskje fått bort kall til stream her også?

Reply:
Logikken ser grei ut. Usikker på hvordan det blir uten stream.